src/components/form/index.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react'
+import Card from '@material-ui/core/Card'
+import CardContent from '@material-ui/core/CardContent'
+import Typography from '@material-ui/core/Typography'
+import { makeStyles } from '@material-ui/core/styles'
+import Button from '@material-ui/core/Button'
+import Box from '@material-ui/core/Box'
+
+import Input from '../input'
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        '& .MuiCardContent-root': {
+            paddingLeft: 0,
+        },
+        marginBottom: '10px',
+        paddingLeft: theme.spacing(2),
+        paddingRight: theme.spacing(2),
+        paddingBottom: theme.spacing(2),
+    },
+    title: {
+        fontSize: 28,
+    },
+}))
+
+const patterns = {
+    discord: /^https:\/\/discord\.com\/api\/webhooks\//,
+    ddb: /^https:\/\/www\.dndbeyond\.com\/profile\/(\w|\d)+\/characters\/\d+$/,
+}
+
+const Form = ({ character = {}, index, onSubmit, handleRemove }) => {
+    const classes = useStyles()
+    const [name, setName] = useState('')
```

Comment:
These [state hooks](https://reactjs.org/docs/hooks-state.html) though 🤯 I'll admit I kinda like the ["declare a class" way](https://reactjs.org/docs/hooks-state.html#equivalent-class-example) better since it is more in tune with my OOP biased mind, but these hooks definitely make things more concise.

Reply:
Get on the train! 🚂